substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -560,19 +566,29 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
-	/// Indices of validators that have offended in the active era and whether they are currently
-	/// disabled.
+	/// Indices of validators that have offended in the active era and their corresponding offence
+	/// (slash percentage).
 	///
 	/// This value should be a superset of disabled validators since not all offences lead to the
 	/// validator being disabled (if there was no slash). This is needed to track the percentage of
 	/// validators that have offended in the current era, ensuring a new era is forced if
 	/// `OffendingValidatorsThreshold` is reached. The vec is always kept sorted so that we can find
 	/// whether a given validator has previously offended using binary search. It gets cleared when
 	/// the era ends.
+	///
+	/// Values of the Vec:
+	/// * u32 - The stash account of the offender
+	/// * Perbill - slash proportion
 	#[pallet::storage]
 	#[pallet::unbounded]
 	#[pallet::getter(fn offending_validators)]
-	pub type OffendingValidators<T: Config> = StorageValue<_, Vec<(u32, bool)>, ValueQuery>;
+	pub type OffendingValidators<T: Config> = StorageValue<_, Vec<(u32, Perbill)>, ValueQuery>;
+
+	/// Keep track which validators are disabled because on new session start they should be
+	/// disabled again. The disabled list in `SessionInterface` is cleared on each new session.
```

Comment:
TODO: https://github.com/paritytech/substrate/pull/14697/files#r1283238804

Reply:
I believe we will always have an upper bound on `DisabledOffenders`. We should store this in a `BoundedVec<u32, MaxDisabledOffenders>`.